src/frontend/src/lib/components/dapps/DappsCarousel.svelte

```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+	import { nonNullish } from '@dfinity/utils';
+	import Carousel from '$lib/components/carousel/Carousel.svelte';
+	import DappsCarouselSlide from '$lib/components/dapps/DappsCarouselSlide.svelte';
+	import {
+		dAppDescriptions,
+		type CarouselSlideOisyDappDescription
+	} from '$lib/types/oisyDappDescription';
+
+	export let styleClass: string | undefined = undefined;
+
+	const dappsCarouselSlides = dAppDescriptions.filter(({ carousel }) =>
+		nonNullish(carousel)
+	) as CarouselSlideOisyDappDescription[];
+</script>
+
+{#if nonNullish(dappsCarouselSlides)}
+	<Carousel {styleClass}>
+		{#each dappsCarouselSlides as dappsCarouselSlide}
```

Comment:
same question about casting here

Reply:
Actually, here it's not casting but `for-each` loop

Reply:
my bad! you are completly right!